### PR TITLE
Updated merge and switchMap to use while loops

### PR DESCRIPTION
**Pull request by rodvieirasilva**

### CODE REVIEW

File: operators.js [hunk 1]
```diff
@@ -59,21 +60,18 @@ const map = (fn) => {
...
+                        return;
+                    }
                     controller.enqueue(value)
-
-                    return read()
                 }
-
-                read()
             }
+            await Promise.all(streams.map(stream => read(stream)))
```
Comment (rodvieirasilva) [inline operators.js:73]: only terminate if all streams are done

File: operators.js [hunk 2]
```diff
@@ -88,24 +86,34 @@ const merge = (streams) => {
  * @return {TransformStream}
  */
 const switchMap = (fn, options = { pairwise: true }) => {
+    let lastRead = undefined;
     return new TransformStream({
         // mousedown
-        transform(chunk, controller) {
+        async transform(chunk, controller) {
```
Comment (rodvieirasilva) [inline operators.js:92]: on all transform calls the inner stream is recreated, like the doc: https://www.learnrxjs.io/learn-rxjs/operators/transformation/switchmap

File: operators.js [hunk 3]
```diff
@@ -24,10 +24,11 @@ const fromEvent = (target, eventName) => {
  */
 const interval = (ms) => {
     let _intervalId
+    let i = 0
     return new ReadableStream({
         start(controller) {
             _intervalId = setInterval(() => {
-                controller.enqueue(Date.now())
+                controller.enqueue(i++, Date.now())
```
Comment (rodvieirasilva) [inline operators.js:31]: add i++ param, like the interval https://www.learnrxjs.io/learn-rxjs/operators/creation/interval

**Comment by ErickWendel** [comment]
Hey @rodvieirasilva thanks for the contribution! Unfortunately I can't edit this source code because It's the source code used on the video tutorial. 

It may be confusing for those who are struggling with syntax or have any bug on their code to see a different implementation looking to the reference.

Thanks anyway and keep rocking!

**Comment by rodvieirasilva** [comment]
Sure, thanks, congratulations for the video.
Regards.

**Pull request closed without merging**